test/cli-alert/src/index.ts

```diff
@@ -10,20 +10,110 @@ if (!process.env.USER_GITHUB_TOKEN || !process.env.SLACK_WEBHOOK_URL) {
 const GITHUB_TOKEN = process.env.USER_GITHUB_TOKEN;
 const SLACK_WEBHOOK_URL = process.env.SLACK_WEBHOOK_URL;
 
-const octokitInstance = new Octokit({
+// Map from job id to it's name and conclusion
+const failedJobs: string[] = [];
+
+const octokit = new Octokit({
   auth: GITHUB_TOKEN,
 });
 const slackWebhook = new IncomingWebhook(SLACK_WEBHOOK_URL);
 
-function filterWorkflows(workflow) {
-  // Keep only workflows didn't succeed, and either completed or were triggered by release
-  return (
-    workflow.conclusion !== 'success' &&
-    (workflow.status === 'completed' || workflow.event === 'release')
+async function checkJobs(smokeTest1: number, smokeTest2: number) {
+  // Get all jobs of the latest 2 smoke tests
+  const jobs1 = (
+    await octokit.actions.listJobsForWorkflowRun({
+      owner: 'snyk',
+      repo: 'snyk',
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      run_id: smokeTest1,
+    })
+  ).data.jobs;
+
+  const jobs2 = (
+    await octokit.actions.listJobsForWorkflowRun({
+      owner: 'snyk',
+      repo: 'snyk',
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      run_id: smokeTest2,
+    })
+  ).data.jobs;
+
+  // If the same job failed in both smoke tests, it has been failing for 2 hours now. Save job to re-run later.
+  for (let i = 0; i < jobs1.length; i++) {
+    if (
+      jobs1[i].conclusion === 'failure' &&
+      jobs1[i].conclusion === jobs2[i].conclusion
+    ) {
+      console.log(`Found a job that failed 2 times in a row: ${jobs1[i].name}`);
+      failedJobs.push(jobs1[i].name);
+    }
+  }
+}
```

Comment:
Comparing by index might be inaccurate here. Don't believe GH guarantees order of the checks. Checking and comparing by names might be safer in this case.

```suggestion
for (const jobName of jobs1.map({name} => name)) {
  if ('failure' === jobs1.find(j => j.name === jobName).conclusion === jobs2.find(j => j.name === jobName).conclusion) {
    failedJobs.push(jobName);
  }
}
```